lib/Models/Terria.ts

```diff
@@ -1241,6 +1241,47 @@ export default class Terria {
     );
   }
 
+  // This method is created to support unit tests.
+  async loadMapItems(model: BaseModel, errors: TerriaError[]) {
+    (await (<any>model).loadMapItems()).pushErrorTo(errors);
+  }
+
+  private async pushAndLoadMapItems(
+    model: BaseModel,
+    newItems: BaseModel[],
+    errors: TerriaError[]
+  ) {
+    if (ReferenceMixin.isMixedInto(model)) {
+      (await model.loadReference()).pushErrorTo(errors);
+
+      if (model.target !== undefined) {
+        await this.pushAndLoadMapItems(model.target, newItems, errors);
+      } else {
+        errors.push(
+          TerriaError.from(
+            "Reference model has no target. Model Id: " + model.uniqueId
+          )
+        );
+      }
+    } else if (GroupMixin.isMixedInto(model)) {
+      (await model.loadMembers()).pushErrorTo(errors);
+
+      model.memberModels.map(async m => {
+        await this.pushAndLoadMapItems(m, newItems, errors);
+      });
+    } else if (MappableMixin.isMixedInto(model)) {
+      newItems.push(model);
+      await this.loadMapItems(model, errors);
+    } else {
+      errors.push(
+        TerriaError.from(
+          "Can not load an un-mappable item to the map. Item Id: " +
+            model.uniqueId
+        )
+      );
+    }
+  }
+
```

Comment:
I think it would be better to add the loading group logic to `workbench.add()` - as it already has the loading reference and mapitems logic

https://github.com/TerriaJS/terriajs/blob/a406adddced1cb994b03bfcdb44748b618cb86ae/lib/Models/Workbench.ts#L161-L212

This would require some changes to `applyInitData`, see next comment

Reply:
The original code also uses loadMapItems logic in `applyInitData` method.